custom_components/tahoma/config_flow.py

```diff
@@ -52,30 +28,29 @@ async def async_step_user(self, user_input=None):
         errors = {}
 
         if user_input is not None:
-            unique_id = user_input.get(CONF_USERNAME)
-            await self.async_set_unique_id(unique_id)
+            username = user_input.get(CONF_USERNAME)
+            password = user_input.get(CONF_PASSWORD)
+
+            await self.async_set_unique_id(username)
             self._abort_if_unique_id_configured()
 
+            client = TahomaClient(username, password)
```

Comment:
It will be nice to have a context manager for TahomaClient, it will avoid to think about the  close:
```Python
with TahomaClient(username, password) as client:
  client.login()
  ....
```

Reply:
Will this auto close / take care of closing on exception? Or is that something that has to be built in the Python API :-).

Reply:
@tetienne nevermind, I see your PR, looks good! Shall we do this improvement in a new PR later on? :)

Reply:
Let's do this in a follow up PR!

Reply:
Yep